components/patient-data/migrations/src/main/java/org/phenotips/data/internal/R54693PhenoTips1500DataMigration.java

```diff
@@ -0,0 +1,141 @@
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.slf4j.Logger;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.objects.DBStringListProperty;
+import com.xpn.xwiki.objects.StringProperty;
+import com.xpn.xwiki.store.XWikiHibernateBaseStore;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.AbstractHibernateDataMigration;
+
+/**
+ * Migration for PhenoTips issue #1500: Automatically migrate {@code global_mode_of _inheritance} from
+ * {@code StringProperty} to {@DBStringListProperty} to allow for the selection of multiple modes
+ * of inheritance.
+ *
+ * @version $Id$
+ * @since 1.2M1
+ */
+@Component
+@Named("R54693PhenoTips#1500")
+@Singleton
+public class R54693PhenoTips1500DataMigration extends AbstractHibernateDataMigration
+{
+    /** Logging helper object. */
+    @Inject
+    private Logger logger;
+
+    /** Resolves unprefixed document names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> resolver;
+
+    /** Serializes the class name without the wiki prefix, to be used in the database query. */
+    @Inject
+    @Named("compactwiki")
+    private EntityReferenceSerializer<String> serializer;
+
+    @Override
+    public String getDescription()
+    {
+        return "Multiple values should be selectable for global mode of inheritance.";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(54693);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), new UpdateModeOfInheritanceCallback());
+    }
+
+    /**
+     * Searches for all documents containing a global_mode_of_inheritance property and changes that property type from
+     * {@code StringProperty} to {@code DBStringListProperty}.
+     */
+    private final class UpdateModeOfInheritanceCallback implements XWikiHibernateBaseStore.HibernateCallback<Object>
+    {
+        /** The names of the properties to fix. */
+        private final String propertyName = "global_mode_of_inheritance";
+
+        @Override
+        public Object doInHibernate(Session session) throws HibernateException, XWikiException
+        {
+
+            XWikiContext context = R54693PhenoTips1500DataMigration.this.getXWikiContext();
+            DocumentReference classReference =
+                new DocumentReference(context.getDatabase(), "PhenoTips", "PatientClass");
+            Query q =
+                session.createQuery("select distinct o.name from BaseObject o, StringProperty p where o.className = '"
+                    + R54693PhenoTips1500DataMigration.this.serializer.serialize(classReference)
+                    + "and p.id.id = o.id and p.id.name =" + this.propertyName + "'");
```

Comment:
Broken query, use this:

    + "' and p.id.id = o.id and p.id.name = '" + this.propertyName + "'");